### PR TITLE
Chore/remove fed functionality

### PR DESCRIPTION
**Pull request by tora-kozic**
### Description of Change ###
Removed FED-related functionality.

### Testing Procedure ###
Run unit/integration tests.

### PR Checklist ###
Did you remember to do the below?

- [x] Add unit tests to verify this change
- [x] Add an entry to CHANGELOG.md describing this change
- [x] Add docstrings for any new public parameters / methods / classes

### CODE REVIEW

**Comment by tora-kozic** [comment]
I have read the CLA Document and I hereby sign the CLA

File: src/py42/clients/authority.py [hunk 1]
```diff
@@ -2,13 +2,5 @@
...
-    [
-        u"administration",
-        u"archive",
-        u"devices",
-        u"legalhold",
-        u"orgs",
-        u"securitydata",
-        u"users",
-    ],
+    [u"administration", u"archive", u"devices", u"legalhold", u"orgs", u"users"],
```
Comment (ryan-haley-code42) [inline src/py42/clients/authority.py:5]: Won't need the `u"`, the merge from master will probably handle that.

**Comment by unparalleled-js** [comment]
`Py42SecurityPlanConnectionError` exception can also be deleted

File: src/py42/clients/securitydata.py [hunk 2]
```diff
@@ -33,145 +28,10 @@ def savedsearches(self):
         """A collection of methods related to retrieving forensic search data.
 
         Returns:
-            :class: `py42._internal.services.securitydata.SavedSearchService`
+            :class: `py42._internal.services.SavedSearchService`
```
Comment (unparalleled-js) [inline src/py42/clients/securitydata.py:31]: I believe this should be `py42._internal.services.savedsearch.SavedSearchService`

File: src/py42/clients/securitydata.py [hunk 3]
```diff
@@ -14,13 +11,11 @@
...
-        security_service,
         file_event_service,
         preservation_data_service,
         saved_search_service,
         storage_service_factory,
     ):
-        self._security_service = security_service
         self._file_event_service = file_event_service
         self._preservation_data_service = preservation_data_service
         self._saved_search_service = saved_search_service
```
Comment (unparalleled-js) [inline src/py42/clients/securitydata.py:21]: Lines 23 and 24 can be removed (as well as the `Lock` import),
it does not seem like they are used anymore

**Comment by timabrmsn** [comment]
I think we can remove `py42.services.storage._auth.SecurityArchiveAuth` and it's usages as well.

File: src/py42/clients/securitydata.py [hunk 4]
```diff
@@ -20,15 +19,13 @@ def __init__(
...
-        self._client_cache = {}
-        self._client_cache_lock = Lock()
 
     @property
     def savedsearches(self):
         """A collection of methods related to retrieving forensic search data.
 
         Returns:
-            :class: `py42._internal.services.SavedSearchService`
+            :class: `py42._internal.services.savedsearch.SavedSearchService`
```
Comment (unparalleled-js) [inline src/py42/clients/securitydata.py:28]: The `_internal` part is still there and not needed (old path)

Comment (tora-kozic) [reply src/py42/clients/securitydata.py:28]: Oh whoops, totally missed that part the first time

**Review by unparalleled-js — APPROVED** [review]

**Review by timabrmsn — APPROVED** [review]

**Pull request merged**